src/index.ts

```diff
@@ -31,49 +33,71 @@ export const isLoggedIn = async (): Promise<boolean> => {
  * @param {AuthTokens} tokens - Access and Refresh tokens
  * @returns {Promise}
  */
-export const setAuthTokens = (tokens: AuthTokens): Promise<void> =>
-  AsyncStorage.setItem(STORAGE_KEY, JSON.stringify(tokens))
+export const setAuthTokens = (tokens: AuthTokens): Promise<void> => {
+  // store accesToken in memory
+  accessToken = tokens.accessToken
+
+  // store refreshToken securely
+  return Keychain.setGenericPassword('refreshToken', tokens.refreshToken, { service: STORAGE_KEY })
+    .then((result) => {
+      if (result) return
+      else throw new Error('Failed to store refresh token')
+    })
+    .catch((error) => {
+      throw error
+    })
+}
 
 /**
  * Sets the access token
  * @async
  * @param {Promise} token - Access token
  */
 export const setAccessToken = async (token: Token): Promise<void> => {
-  const tokens = await getAuthTokens()
-  if (!tokens) {
+  const refreshToken = await getRefreshToken()
+  if (!refreshToken || !accessToken) {
```

Comment:
should it be `!refreshToken || !token`?

Reply:
Both tokens need to be checked, like before.